scripts/python-skywater-pdk/skywater_pdk/fet_simulator/generate-docs-plots.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 SkyWater PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+from pathlib import Path
+
+from fet_simulator import generate_fet_plots
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'fd_pr_dir',
+        help='Path to the particular version of the primitive library',
+        type=Path
+    )
+    parser.add_argument(
+        'device_details_dir',
+        help='Path to the directory with device details to save images to',
+        type=Path
+    )
+
+    args = parser.parse_args()
+
+    typicalcorner = args.fd_pr_dir / 'models/corners/tt.spice'
+
+    fets = [
```

Comment:
Is there any reason this couldn't be auto-generated by just listing the directory with `*fet*`?

Reply:
When I was writing the script I was relying on the directory names and structure from https://github.com/google/skywater-pdk/tree/master/docs/rules/device-details - the `generate-docs-plots.py` saves SVG files in such a manner that they go to the corresponding FET directories in the Device details. In this directory there were some directories that were named differently in comparison to names in `sky130_fd_pr`, e.g. `nfet_20v0_nvt_iso` from sky130_fd_pr is described in `nfet_20v0_iso` directory, both `nfet_05v0_nvt` and `nfet_03v3_nvt` from sky130_fd_pr are described in `nfet_03v3_vnt-and-nfet_05v0_nvt` directory in Device details.

I can change it so it will scan sky130_fd_pr directory for FET cells and produce directories with FET names and corresponding SVG files in those directories.

Reply:
We should fix it so that the device-details directories match the real device names.

Reply:
I changed the script so it now scans for FET cells in libraries. If the script fails to generate the plot for FET cell it saves the path to the failing bins.csv file to the file given in `--failed-inputs` flag. The SVG files are now saved in such a way that for each:
```
libraries/<library>/<version>/cells/<fet_name>/<library>__<fet_name>.bins.csv
```
file there will be corresponding:
```
output/<library>/<version>/cells/<fet_name>/<library>__<fet_name>__<fet_modelname>_W<width_value>.sim.svg
```
where:
- `library` is the name of the library (FETs are in general located only in `sky130_fd_pr`)
- `fet_name` is FET cell name
- `fet_modelname` is the FET model name (present in Spice files and bins.csv files)
- `width_value` is the width value for which the simulation was performed, with underscore in place of decimal.

The example names are:
```
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_fT_W20_35.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_fT_W40_31.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_fT_W5_4.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_gm_gds_W20_35.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_gm_gds_W40_31.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_gm_gds_W5_4.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_gm_id_W20_35.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_gm_id_W40_31.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_gm_id_W5_4.sim.svg
sky130_fd_pr/latest/cells/esd_nfet_01v8/sky130_fd_pr__esd_nfet_01v8__sky130_fd_pr__esd_nfet_01v8_Id_w_W20_35.sim.svg
```